api/types/access_request.go

```diff
@@ -94,6 +94,8 @@ type AccessRequest interface {
 	GetRequestedResourceIDs() []ResourceID
 	// SetRequestedResourceIDs sets the resource IDs to which access is being requested.
 	SetRequestedResourceIDs([]ResourceID)
+	// GetNodeResourceIDsByCluster returns a map of node resource IDs by cluster name
+	GetNodeResourceIDsByCluster() map[string][]ResourceID
```

Comment:
Does this need to be a method on `AccessRequest`?

Looks like it can be implemented purely by using the existing `GetRequestedResourceIDs`.

Reply:
The method was my suggestion. I'm not sure if that is best, my thought was that it is just returning data from the access request and is used in multiple packages, it could be a method. It could just be a function somewhere too, doesn't make much difference to me

Reply:
Gotcha. 

In this case, a free function is much preferred. 

If we add a new implementation of the access request interface, we don't want to force ourselves to reimplement this method since it doesn't depend on the implementation in any way. 

Another way to think about it - does this method operate on an access request (func) or does it define the behavior of an access request (interface method)?

Reply:
Sounds fine to me. I moved it back into `services` 👍